client/core/core.go

```diff
@@ -5288,114 +5396,142 @@ func (c *Core) prepareTrackedTrade(dc *dexConnection, form *TradeForm, crypter e
+	err := tr.dc.signAndRequest(tr.msgOrder, tr.route, result, fundingTxWait+DefaultResponseTimeout)
 	if err != nil {
 		// At this point there is a possibility that the server got the request
 		// and created the trade order, but we lost the connection before
 		// receiving the response with the trade's order ID. Any preimage
 		// request will be unrecognized. This order is ABANDONED.
-		return nil, nil, fmt.Errorf("new order request with DEX server %v market %v failed: %w", dc.acct.host, mktID, err)
+		return nil, fmt.Errorf("new order request with DEX server %v market %v failed: %w", tr.dc.acct.host, tr.mktID, err)
 	}
 
 	// If we encounter an error, perform some basic logging.
-	logAbandon := func(err interface{}) {
+	logAbandon := func(err error) {
 		c.log.Errorf("Abandoning order. preimage: %x, server time: %d: %v",
-			preImg[:], result.ServerTime, err)
+			tr.preImg[:], result.ServerTime, err)
 	}
 
-	err = validateOrderResponse(dc, result, ord, msgOrder) // stamps the order, giving it a valid ID
+	ord := tr.dbOrder.Order
+	err = validateOrderResponse(tr.dc, result, ord, tr.msgOrder) // stamps the order, giving it a valid ID
 	if err != nil {
-		logAbandon(fmt.Sprintf("order response validation failure: %v", err))
-		return nil, nil, fmt.Errorf("validateOrderResponse error: %w", err)
+		logAbandon(fmt.Errorf("validateOrderResponse error: %v", err))
+		return nil, fmt.Errorf("order response validation failure: %v", err)
 	}
 
 	// TODO: Need xcWallet fields for acceptable SwapConf values: a min
 	// acceptable for security, and even a max confs override to act sooner.
 
 	// Store the order.
-	dbOrder := &db.MetaOrder{
-		MetaData: &db.OrderMetaData{
-			Status: order.OrderStatusEpoch,
-			Host:   dc.acct.host,
-			Proof: db.OrderProof{
-				DEXSig:   result.Sig,
-				Preimage: preImg[:],
-			},
-			EpochDur:           mktConf.EpochLen, // epochIndex := result.ServerTime / mktConf.EpochLen
-			FromSwapConf:       assetConfigs.fromAsset.SwapConf,
-			ToSwapConf:         assetConfigs.toAsset.SwapConf,
-			MaxFeeRate:         assetConfigs.fromAsset.MaxFeeRate,
-			RedeemMaxFeeRate:   assetConfigs.toAsset.MaxFeeRate,
-			FromVersion:        assetConfigs.fromAsset.Version,
-			ToVersion:          assetConfigs.toAsset.Version, // and we're done with the server's asset configs.
-			Options:            form.Options,
-			RedemptionReserves: redemptionReserves,
-			RefundReserves:     refundReserves,
-			ChangeCoin:         changeID,
-			ProgramID:          form.Program,
-		},
-		Order: ord,
+	tr.dbOrder.MetaData.Status = order.OrderStatusEpoch
+	tr.dbOrder.MetaData.Proof = db.OrderProof{
+		DEXSig:   result.Sig,
+		Preimage: tr.preImg[:],
 	}
-	err = c.db.UpdateOrder(dbOrder)
+
+	err = c.db.UpdateOrder(tr.dbOrder)
 	if err != nil {
-		logAbandon(fmt.Sprintf("failed to store order in database: %v", err))
-		return nil, nil, fmt.Errorf("Order abandoned due to database error: %w", err)
+		logAbandon(fmt.Errorf("UpdateOrder error: %w", err))
```

Comment:
This message is now less informative. Before it was at least clear it was the DB.  Let's leave both `logAbandon` calls the same please, messages and Sprintf... or remove `logAbandon` as per my other comment.